src/main/scala/scalikejdbc/bigquery/Extractor.scala

```diff
@@ -4,9 +4,20 @@ import scalikejdbc._
 
 class Extractor[A](statement: SQLSyntax, f: WrappedResultSet => A) {
 
-  def list: Runner[Seq[A]] = Runner(statement)(_.map(f)(collection.breakOut))
+  def iterator: Runner[Iterator[A]] = Runner(statement)(_.map(f))
 
-  def single: Runner[Option[A]] = Runner(statement)(_.map(f).headOption)
+  def list: Runner[Seq[A]] = Runner(statement)(_.map(f).toList)
+
+  def single: Runner[Option[A]] = Runner(statement) { rs =>
+    val rows = (rs map f).toList
```

Comment:
`toList` can be inefficient if ResultSet contains many rows.
How about using `toStream` ?